### PR TITLE
[SPARK-29444][FOLLOWUP] add doc and python parameter for ignoreNullFields in json generating

### PR DESCRIPTION
**Pull request by jackylee-ch**
# What changes were proposed in this pull request?
Add description for ignoreNullFields, which is commited in #26098 , in DataFrameWriter and readwriter.py.
Enable user to use ignoreNullFields in pyspark.

### Does this PR introduce any user-facing change?
No

### How was this patch tested?
run unit tests

### CODE REVIEW

**Comment by stczwd** [comment]
cc @HyukjinKwon @cloud-fan

**Comment by HyukjinKwon** [comment]
ok to test

File: python/pyspark/sql/readwriter.py [hunk 1]
```diff
@@ -788,7 +788,7 @@ def saveAsTable(self, name, format=None, mode=None, partitionBy=None, **options)
 
     @since(1.4)
     def json(self, path, mode=None, compression=None, dateFormat=None, timestampFormat=None,
-             lineSep=None, encoding=None):
+             lineSep=None, encoding=None, ignoreNullFields=None):
```
Comment (dongjoon-hyun) [inline python/pyspark/sql/readwriter.py]: This seems to be beyond `add doc`. Could you update the PR title, @stczwd ?

Comment (stczwd) [reply python/pyspark/sql/readwriter.py]: Okey

File: sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala [hunk 2]
```diff
@@ -1189,9 +1189,10 @@ object SQLConf {
 
   val JSON_GENERATOR_IGNORE_NULL_FIELDS =
     buildConf("spark.sql.jsonGenerator.ignoreNullFields")
-      .doc("If false, JacksonGenerator will generate null for null fields in Struct.")
-      .stringConf
-      .createWithDefault("true")
+      .doc("Whether to ignore null fields in column/struct during json generating. " +
+        "If false, json generator will generate null in Column/Struct.")
```
Comment (HyukjinKwon) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala]: `Column/Struct` -> `JSON objects`

File: sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala [hunk 3]
```diff
@@ -1189,9 +1189,10 @@ object SQLConf {
 
   val JSON_GENERATOR_IGNORE_NULL_FIELDS =
     buildConf("spark.sql.jsonGenerator.ignoreNullFields")
-      .doc("If false, JacksonGenerator will generate null for null fields in Struct.")
-      .stringConf
-      .createWithDefault("true")
+      .doc("Whether to ignore null fields in column/struct during json generating. " +
```
Comment (HyukjinKwon) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala]: I would just write like this:

```
Whether to ignore null fields when generating JSON objects in JSON data source and 
JSON functions such as to_json.
If false, it generates null for null fields in JSON objects.
```

File: sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala [hunk 4]
```diff
@@ -687,6 +687,8 @@ final class DataFrameWriter[T] private[sql](ds: Dataset[T]) {
    * <li>`encoding` (by default it is not set): specifies encoding (charset) of saved json
    * files. If it is not set, the UTF-8 charset will be used. </li>
    * <li>`lineSep` (default `\n`): defines the line separator that should be used for writing.</li>
+   * <li>`ignoreNullFields` (default `true`): whether to ignore null fields in column/struct
+   * during json generating. </li>
```
Comment (HyukjinKwon) [inline sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala]: here too

**Review by HyukjinKwon — APPROVED** [review]
Looks good otherwise.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala (hunk 3 above)
Comment (stczwd) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala]: Good, it's better.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Merged to master.

**Pull request closed without merging**